be/src/exec/pipeline/fragment_executor.cpp

```diff
@@ -706,11 +709,67 @@ Status FragmentExecutor::_decompose_data_sink_to_operator(RuntimeState* runtime_
             fragment_ctx->pipelines().emplace_back(pp);
         }
     } else if (typeid(*datasink) == typeid(starrocks::stream_load::OlapTableSink)) {
-        runtime_state->set_per_fragment_instance_idx(request.sender_id());
+        size_t desired_tablet_sink_dop = request.pipeline_sink_dop();
+        DCHECK(desired_tablet_sink_dop > 0);
```

Comment:
Under what instance the two numbers are not equal?

Reply:
If they are not equal, how to guide the user to improve the parallelism by `set variables`?

Reply:
For insert into select statement, we can't know the dop of tablet sink in advance.
Such as user may add a limit operator at the end of select statement, then the dop is 1. If we set table_sink_dop to be 8, then this two numbers are not equal.

Reply:
If we will use the select dop to set the sink dop, there is not necessary to define a variable pipelineSinkDop.